src/flow_dot_js.ml

```diff
@@ -224,6 +224,29 @@ let infer_type filename content line col =
       (None, Some (loc, ty, raw_type, reasons))
     | _, _ -> failwith "parse error"
 
+let dump_types js_file js_content =
+    let filename = Loc.SourceFile (Js.to_string js_file) in
+    let root = Path.dummy_path in
+    let content = Js.to_string js_content in
+    match parse_content filename content with
+    | ast, [] ->
+      (* defaults *)
+      let metadata = stub_metadata ~root ~checked:true in
+
+      Flow_js.Cache.clear();
+
+      let cx = Type_inference_js.infer_ast
+        ~metadata ~filename ~module_name:(Modulename.String "-") ast
+      in
+      let printer = Type_printer.string_of_t in
+      let raw_printer _ _ = None in
+      let types = Query_types.dump_types printer raw_printer cx in
+
+      let strip_root = None in
+      let types_json = DumpTypesCommand.types_to_json types ~strip_root in
+
+      js_of_json types_json
+    | _, _ -> failwith "parse error"
```

Comment:
That's the same messages used by `infer_type` when `parse_content` fails: https://github.com/zertosh/flow/blob/666003571ef634f186056891ddaf2287741906cd/src/flow_dot_js.ml#L225